BUILD.gn

```diff
@@ -228,6 +233,45 @@ run_node("deno_runtime_declaration") {
   }
 }
 
+# Generates the worker TypeScript type library for deno that will be
+# included in the worker runtime bundle
+run_node("deno_worker_runtime_declaration") {
```

Comment:
Is it necessary to have a different bundle, declaration, and snapshot? It would be nice to just use the main one for now... I realize there is slightly different semantics between the main isolate and workers, but for the time being I think we should ignore that.

Reply:
This would leave the main isolate with global methods that cause a panic during dispatch or type checks that are not really safe. I don't really like either of the alternatives, and this also gives the ability to add worker specific APIs or implementations. My thought was that after #1739 the native TS compiler could be built just like any other third party compiler, and snapshots for all compilers could be cached on first run eliminating the need for the compiler snapshot.

Reply:
> This would leave the main isolate with global methods that cause a panic during dispatch or type checks that are not really safe.

I'd rather that for now than adding another snapshot/bundler. We're refactoring this whole bit right now and I want to err on the side of simplicity. I also think it's unnecessarily increasing the executable size. I'm sure we can work around this somehow in the future.